pkg/build/registry/buildconfig/webhook.go

```diff
@@ -62,12 +62,37 @@ func (c *controller) ServeHTTP(w http.ResponseWriter, req *http.Request, ctx kap
 		return nil
 	}
 
+	buildTriggerCauses := generateBuildTriggerInfo(revision, hookType)
 	request := &buildapi.BuildRequest{
-		ObjectMeta: kapi.ObjectMeta{Name: name},
-		Revision:   revision,
+		TriggeredBy: buildTriggerCauses,
+		ObjectMeta:  kapi.ObjectMeta{Name: name},
+		Revision:    revision,
 	}
+
 	if _, err := c.instantiator.Instantiate(config.Namespace, request); err != nil {
 		return errors.NewInternalError(fmt.Errorf("could not generate a build: %v", err))
 	}
 	return nil
 }
+
+func generateBuildTriggerInfo(revision *buildapi.SourceRevision, hookType string) (buildTriggerCauses []buildapi.BuildTriggerCause) {
+	switch {
+	case hookType == "generic":
```

Comment:
i know, when i tried to use it here it started exploding because the types don't match. but maybe i did something wrong?
it works ok in other places.